### ISSUE TITLE
Make it easy to set MADCTL for rotation

### ISSUE THREAD

**Issue by jepler** [adafruit/Adafruit_CircuitPython_ILI9341#26]
The upper 3 bits of the MADCTL register can set a rotation that occurs within the LCD controller.  Particularly when working with OnDiskBitmaps, displayio can be much more efficient when the displayio rotation is 0.  By exposing the ability to modify the initialization string's MADCTL value, the more efficient in-controller rotation can be used instead.

Locally, I am using this code, but it's a bit gross:
```python
def init_with_madctl(madctl):
    return adafruit_ili9341._INIT_SEQUENCE.replace(b'\x36\x01\x38',
            bytes([0x36, 1, madctl]))
…
display = displayio.Display(display_bus,
        init_with_madctl(0b010_110_00), width=320, height=240)
```

Section 9.3 and the figure on page 209 of https://cdn-shop.adafruit.com/datasheets/ILI9341.pdf show how these 3 bits of MADCTL are interpreted.

**Comment by makermelissa** [comment]
Why are you using MADCTL for rotation? CircuitPython handles the rotation itself and then draws in the normal orientation. I believe this is to help with screen tearing according to @tannewt.

**Comment by jepler** [comment]
Using MADCTL for rotation usually performs better, especially for OnDiskBitmaps where displayio rotation gives pathologically poor behavior.

**Comment by tannewt** [comment]
I never did the full tearing work I intended to do. So I wouldn't block this on that goal. Tearing happens with CP now and there isn't much we can do.

**Comment by makermelissa** [comment]
Fair enough. Should be easy enough to refactor this driver to allow easier modification of MADCTL.

**Comment by ritesh006** [comment]
Please review this commit
